app/flatpak-builtins-install.c

```diff
@@ -300,10 +301,30 @@ flatpak_builtin_install (int argc, char **argv, GCancellable *cancellable, GErro
   if (!flatpak_split_partial_ref_arg (name, &opt_arch, &branch, error))
     return FALSE;
 
+  /* Make sure the default branch is pulled from the server's summary
+     file if it's defined there and hasn't been pulled in yet. */
+  default_branch = flatpak_dir_get_remote_default_branch (dir, repository);
```

Comment:
I wonder if this is the right behaviour. If we anyway are downloading the remote summary to get the default branch every time, then maybe we should not store the default locally anyway? That way we can follow changes in the upstream default branch unless you manually set a local value.

Hmm, although it may be nice to have a local version of this property for use in e.g. gnome-software when offline.